src-electron/upgrade/upgrade.js

```diff
@@ -40,14 +48,48 @@ async function getForcedExternalStorage(db, attributeId) {
   let pkgs = await queryPackage.getPackageRefByAttributeId(db, attributeId)
   let zcl = await queryPackage.getPackageByPackageId(db, pkgs)
   zcl = zcl.path
-  let obj = await fsp.readFile(zcl)
-  let data = JSON.parse(obj)
+  let obj = await fsp.readFile(zcl, 'utf-8')
+  let data
+  if (isJsonString(obj)) {
+    data = JSON.parse(obj)
+  }
   let byName = data?.attributeAccessInterfaceAttributes
   let lists = data?.listsUseAttributeAccessInterface
   let forcedExternal = { byName, lists }
   return forcedExternal
 }
 
+/**
+ * Returns a list of attributes with proper global attributes
+ *
+ * @export
+ * @param {*} db
+ * @param {*} clusterRef
+ * @param {*} attributes
+ * @returns attributes
+ */
+
+async function computeStorageTemplate(db, clusterRef, attributes) {
+  let clusterName
+  let forcedExternal
+  clusterName = await queryCluster.selectClusterName(db, clusterRef)
+  return await Promise.all(
+    await attributes.map(async (attribute) => {
```

Comment:
Why do you need the "await attributes" bit?  `attributes.map(async ... )` should give you an array of promises, which is what you want to pass to Promise.all.  Here you're taking the array, creating a promise from it, then awaiting the promise, which adds one extra trip through the event loop, then getting the resulting array and passing it to `all()`... why the extra promise in there?

Reply:
This is not needed. Thank you. Will remove.

Reply:
@bzbarsky-apple this is the only change I did not push because taking out "await" causes Zigbee generation to fail. I *think* I know why but will do more debug tomorrow and get back to you with a more precise answer. Thank you.

Reply:
added in separate commit to track CI https://github.com/project-chip/zap/actions/runs/6490297672/job/17625855543

Reply:
@bzbarsky-apple my understanding is that this "await" works as a "try/catch" equivalent. The code fails CI without it. That said, I will debug this more tomorrow and get back to you with more detail.

Reply:
removed